### PR TITLE
Issue #13100: Create tests from AnnotationLocation xdoc examples

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #13100

https://checkstyle.sourceforge.io/checks/annotation/annotationlocation.html#AnnotationLocation

### CODE REVIEW

File: src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationExamplesTest.java
Comment (stoyanK7) [inline src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationExamplesTest.java:27]: I created #13345. I will update description after this PR gets merged.

File: src/xdocs/checks/annotation/annotationlocation.xml.template
Comment (stoyanK7) [inline src/xdocs/checks/annotation/annotationlocation.xml.template:143]: I am unsure about making those templates with examples now. If a change to some xdoc has to be made, it has to be made here too. 

What we can do is add those `id="Example1.config"` anchors and later generate the template from them. Just pick the `source` after the `p` and swap the contents with the value of the `id`.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/Example2.txt
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/Example2.txt:2]: Add Checker

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs/checks/annotation/annotationlocation.xml.template
Comment (nrmancuso) [reply src/xdocs/checks/annotation/annotationlocation.xml.template:143]: > I am unsure about making those templates with examples now. If a change to some xdoc has to be made, it has to be made here too.

Please provide some details or an example, I am not sure exactly what this means.

Comment (stoyanK7) [reply src/xdocs/checks/annotation/annotationlocation.xml.template:143]: What I mean is, it will take a month before we start generating xdocs from templates. If an example is added or documentation is written in that time(1 month), both the template and the original file have to be changed.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/Example2.txt
Comment (stoyanK7) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/Example2.txt]: Done.

File: src/xdocs/checks/annotation/annotationlocation.xml.template
Comment (romani) [reply src/xdocs/checks/annotation/annotationlocation.xml.template:143]: original file like https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/checks/annotation/annotationlocation.xml stays in git.
we create another file that is in template.
each time `mvn test` are executing, there some Test class, that actually not a test but just executes by junit, this "Test" generates src/xdocs/checks/annotation/annotationlocation.xml from template, override file that is in git. This "Test" does not touch any other xdoc files if template is not present.

if somebody will update xdoc and forget to update template, it will be visible in git diff, and be blocked in PR review.

this will help us module by module migrate to templates (just new files are created). No blockers to other, release at any time, invisible to others, any regression on examples parser or ..... will be detected immediately by unexpected git diff.

as all are migrated and we have time and we really want to remove static xdoc modules files.
We can start final phase is to figure out how to remove xdoc static files and play with site phase to generate files to special folder and site phase will take files from generated area and combine with static xdoc (not for modules) pages. There are a lot of unknown here for me on how to do this. I have no idea who to do this to be honest, but it might be possible.

Comment (romani) [reply src/xdocs/checks/annotation/annotationlocation.xml.template:143]: On a moment of finishing template for module and having all examples in separate files, we need to remove this module from validation of javadoc-xdoc to have same content.

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by stoyanK7** [comment]
Moving to draft. Until #13357 is merged.

**Comment by romani** [comment]
@stoyanK7 , what blocks this PR is draft mode ?

**Comment by stoyanK7** [comment]
The templates will be using macros instead of the placeholders (`${Example...}`) I initially used here.

**Comment by romani** [comment]
No need to do templates for all/many modules, please use only one. And let's finish draft implementation on it, each new module template might improve implementation more and more.
We still migration options to do migration module by module in separate PRs.

**Comment by romani** [comment]
@stoyanK7 , what blocks this PR to me merged?

**Review by nrmancuso — APPROVED** [review]
Ok to merge as CI passes

**Pull request merged**
